### ISSUE TITLE
Optional UI sound effects

### ISSUE THREAD

**Issue by quincylvania** [openstreetmap/iD#5821]
Many apps use subtle sound effects to reinforce their interfaces and give audial feedback. Messaging apps bloop upon sending, video and sound editing apps click when snapping clips end-to-end, and various apps thunk upon errors. Games widely use sound effects for all sorts of input like button presses and focus changes.

There are various places in iD where I think minor, tasteful sound effects could create a richer editing experience and help the vision-impaired.

- Click upon snapping to a node or way when moving or drawing vertices
- Snip upon splitting a way
- Error thunk upon trying to create a self-intersecting way
- Selection sound upon picking a preset
- Cheerful completion sound upon successful changeset upload
- More…

Of course users would be able to mute sounds at any time.

**Comment by mmd-osm** [comment]
Yeah, back in the days, Potlatch 0.10c had a nifty beep sound when saving a changeset. That was like 11 years ago. 

And it's still around ~~today~~: https://github.com/openstreetmap/openstreetmap-website/blob/7bd1cda/public/potlatch/beep.mp3

https://github.com/openstreetmap/potlatch/blame/40b208026d0fe3261856c0fb5cba4266dbd1147c/potlatch.as#L40-L42

**Comment by antimundo** [comment]
I think UI sounds can give a lot of feedback to the user and clarify what is happening, but if something like this was implemented, I would do add it with the purpose of been useful and provide feedback while trying not be very intrusive, I don't like the approach of adding sounds that makes the editor sound like a video game.

Also if implemented, these sound effects should be optional.
